syntax = "proto3";

package Speechpro.Cloud.ASR;

/**
* Описание модели
*/
message Model
{
	// Уникальный идентификатор
	string id = 1;
	// Количество активных сессий
	uint64 active_sessions = 2;
}

/**
* Список доступных моделей
*/
message ListOfModels
{
	repeated Model models = 1;	
}

/**
* Финализирующее сообщение.
* Сигнал сервису, что в рамках данной сессии пользователь больше не будет отправлять новые данные и ожидает всех результатов распознавания.
*/
message Finish
{ 

}

/**
* Текст из нескольких слов
*/
message Text
{
	repeated Word words = 1;
}

/**
* Слово
*/
message Word
{
	// Текст слова
	string text = 1;
	// Временная метка начала слова (мс)
	int64 start_ms = 2;
	// Длина произнесенного слова (мс)
	int64  length_ms = 3;
	// Мера уверенности в распознанном слове [0.0; 1.0]
	float confidence = 4;

	// Поддерживаемые знаки препинания
	enum PunctuationMark
	{
		NONE = 0;				// нет знака препинания
		DOT = 1;				// точка
		COMMA = 2;				// запятая
		COLON = 3;				// двоеточие
		SEMICOLON = 4;			// точка с запятой
		EXCLAMATION_POINT = 5;	// восклицательный знак
		QUESTION_MARK = 6;		// вопросительный знак
		DASH = 7;				// тире
		SPEAKER_CHANGE = 8;		// точка смены диктора (. -)
	}
	// Знак препинания, стоящий после распознанного слова
	PunctuationMark punctuation_mark = 5;

	// Информация о дикторе, произнесшем это слово
	DictorInfo dictor = 6;
}

/**
* Информация о дикторе
*/
message DictorInfo
{
	// Информация о дикторе не всегда доступна
	oneof dictor_info
	{
		// Не удалось определить диктора
		DictorNotDetected dictor_not_detected = 1;
		// Диктор определен. Его номер.
		int32 dictor_num = 2;
	}
	// Причины, по которым диктор не определен
	enum DictorNotDetected
	{
		UNKNOWN_DICTOR = 0; // Диктор не определен
	}
}

/**
* Звук
*/
message Sound
{
	// Байты сэмплов звука. Рекомендуется подавать звук порциями до 60 секунд.
	// Внимание: поддерживается только pcm mono 16bit signed!!!
	bytes samples = 1;
}

